### PR TITLE
Move the reader-osm module into core

### PR DESCRIPTION
**Pull request by easbar**
Most of the content of this module has already been moved into core anyway. Users who used `graphhopper-reader-osm` as dependency in their project should now use `graphhopper-core` instead.

### CODE REVIEW

**Review by karussell — APPROVED** [review]
Thanks, LGTM!

File: core/files/changelog.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 3.0
+    moved the reader-osm module into core, GraphHopperOSM is deprecated, use GraphHopper instead
```
[the remark this answers is not in the document]
Comment (easbar) [reply core/files/changelog.txt:2]: It's been deprecated for some time already

[the remark this answers is not in the document]
Comment (karussell) [reply core/files/changelog.txt]: Ah, I see ... maybe then just remove it then :) ?

[the remark this answers is not in the document]
Comment (easbar) [reply core/files/changelog.txt]: It was not deprecated in the last release so I though I just leave it until after 3.0, but we can also remove it now, don't know.

**Pull request merged**